collectors/feed/collectors.py

```diff
@@ -0,0 +1,359 @@
+
+        logger.info("Fetching new feeds")
+        start_dt = timezone.now()
+
+        feed_data = self.configuration_adapter.get_config()
+
+        total, allowed = 0, 0
+        for single_feed in feed_data:
+            incr_total, incr_allowed = self.collect_feed(single_feed)
+            total += incr_total
+            allowed += incr_allowed
+
+        self.store(updated_until_dt=start_dt)
+
+        msg = f"{self.name} is updated until {start_dt}."
+        msg += f" Total new Feed entries fetched: {total}"
+        msg += f" Allowed new Feed entries fetched: {allowed}"
+
+        logger.info("Feed sync finished.")
+        return msg
+
+    @staticmethod
+    def collect_feed(feed_data: dict) -> Tuple[int, int]:
+        """Scrape a particular feed."""
+
+        feed_name = feed_data["name"]
+        url = feed_data["url"]
+        entry_url_replace = feed_data.get("entry_url_replace", [])
+        entry_separate_links = feed_data.get("entry_separate_links")
+        skip_entry_title_regex = feed_data.get("skip_entry_title_regex")
+        entry_text_in_html = feed_data.get("entry_text_in_html")
+        components = feed_data.get("components", [])
+
+        raw_feed_entries = FeedCollector.fetch_new_feed_entries(url, feed_name)
+
+        pruned_raw_feed_entries = FeedCollector.prune_new_raw_feed_entries(
+            raw_feed_entries, skip_entry_title_regex
+        )
+
+        raw_entries_w_summaries = FeedCollector.get_summaries_for_raw_feed_entries(
+            pruned_raw_feed_entries,
+            entry_url_replace,
+            entry_separate_links,
+            entry_text_in_html,
+        )
+
+        filtered_raw_entries_w_summaries = (
+            FeedCollector.filter_feed_entries_with_summaries(raw_entries_w_summaries)
+        )
+
+        content_dicts_for_allowed_snippets = (
+            FeedCollector.prepare_data_for_eligible_feed_entries(
+                filtered_raw_entries_w_summaries, components
+            )
+        )
+
+        all_urls = set()
+        for entry in raw_feed_entries:
+            all_urls.add(entry.link)
+
+        allowed_snippet_urls = set()
+        for content in content_dicts_for_allowed_snippets:
+            allowed_snippet_urls.add(content["url"])
+            new_snippet = Snippet(
+                source=feed_name,
+                content=content,
+            )
+            logger.info(
+                "Creating allowed Snippet: %s", repr(new_snippet.content["url"])
+            )
+            logger.debug("new allowed Snippet data: %s", repr(new_snippet.__dict__))
+            # Save separately so that log shows the context if it fails.
+            new_snippet.save()
+
+        # Prevent reprocessing (or fetching) the same entries in future Collector runs.
+        new_blocked_snippet_urls = all_urls - allowed_snippet_urls
+        for url in new_blocked_snippet_urls:
+            content = {
+                "url": url,
+                "create_flaws": False,
```

Comment:
Makes sense. If we go without the switch as you describe, then it will be crucial for the collector itself to not hand over the would-be-`"create_flaws": False` Snippets to the Flaw creator, right?

If true, then I'll add a TODO to the code to send only `"create_flaws": True` to the Flaw creator, once it starts to exist, and to eliminate `create_flaws` if possible (the Flaw creator will ignore it anyway). Sound right?

The Snippet-to-Flaw creator doesn't exist yet AFAIK, so that is probably the next best thing.

Reply:
Actually you have a good point. We can leave it on the collector at this point when it is just RSS feed collector and eventually turn it into a proper attribute if we encounter another like this. For the time being your collector would just not call the flaw creation logic.